content/articles/How-to-start-up-with-Socket-Programming-with-Java/index.md

```diff
@@ -0,0 +1,331 @@
+
+This is an article about Sockets. It will give you a glimpse of how clients and servers communicate over the internet.
+
+Table of content:
+- [What is a socket](#what-is-a-socket)
+- [Programming on Server-side](#programming-on-server-side)
+- [](#)
+- [](#)
+- [](#)
+
+### Introduction
+Socket programming is a way of connecting two computers on a network to allow the exchange of information between the two machines that are running over `JRE`(Java Runtime Environment). The connection can be through **connection-oriented protocol** or **connectionless** but for our case, we will be using connection-oriented which is TCP/IP.
+
+TCP is a connection-oriented protocol, which means that before data can be transferred, two computers must first establish a connection.
+
+Connectionless -For this type of connection, UDP protocol (User Datagram Protocol)also a layer 2 protocol, is used to provide a good picture where you do not necessarily need every packet to arrive at its desired destination.
+
+We will be using Client\Server architecture to illustrate more on sockets. Client and server communicate by writing to and reading from socket connection thus for a connection to happen between the server and a client, a socket must be present. The two machines should know each other's network information.
```

Comment:
I would suggest rephrasing this part to something like: 

" We will be using a Client\Server architecture to illustrate more on sockets. The client and server will communicate by writing to and reading from a socket connection. To establish this connection the machines must have each other's network information."

 I felt like there needed to be a smoother transition into the third sentence there it felt slightly awkward and I believe grammar wise you were missing "the" in front of "client and server" since they are nouns.